### PR TITLE
B leauge stats

### PR DESCRIPTION
**Pull request by brian-greeson**
Iteration 3 Functionality

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -24,6 +24,233 @@ def self.create_items(file, item_class)
...
+ def gameteams_matching_games(games)
+   GameTeam.all.select do |game_id, gameteam|
+     games.keys.include?(game_id)
+   end
+ end
+
+  def coaches_with_team_id(games)
+    # get a list of gameteams in the season
+    gamesteams = gameteams_matching_games(games)
+    # get list of coaches in the current season
```
Comment (DavidTTran) [inline lib/stat_tracker.rb:224]: Need to remove comments in the final refactor.

**Review by DavidTTran — APPROVED** [review]
I believe there's still going to be an issue regarding who used the sample fixtures and who didn't. But good job!

**Comment by kristastadler** [comment]
Did group review of pull request.

This pull request adds iteration three functionality, including:

- count_of_teams (Integer)
- best_offense (String
- worst_offense (String)
- best_defense (String)
- worst_defense (String)
- highest_scoring_visitor (String)
- highest_scoring_home_team (String)
- lowest_scoring_visitor (String)
- lowest_scoring_home_team (String)
- winningest_team (String)
- best_fans (String)
- worst_fans (Array)

Also made the decision to change stat_tracker_test.rb to use full data instead of fixtures.

Will probably need to remove duplicate games_in_season method, since Iteration 2 also uses the method and was added previously.

**Comment by kristastadler** [comment]
Did group review of pull request.

This pull request adds iteration three functionality, including:

- count_of_teams (Integer)
- best_offense (String
- worst_offense (String)
- best_defense (String)
- worst_defense (String)
- highest_scoring_visitor (String)
- highest_scoring_home_team (String)
- lowest_scoring_visitor (String)
- lowest_scoring_home_team (String)
- winningest_team (String)
- best_fans (String)
- worst_fans (Array)

Also made the decision to change stat_tracker_test.rb to use full data instead of fixtures.

Will probably need to remove duplicate games_in_season method, since Iteration 2 also uses the method and was added previously.

**Pull request merged**